R/froll.R

```diff
@@ -1,11 +1,42 @@
-froll = function(fun, x, n, fill=NA, algo=c("fast", "exact"), align=c("right", "left", "center"), na.rm=FALSE, hasNA=NA, adaptive=FALSE) {
+partial2adaptive = function(x, n, align) {
+  if (align=="center")
+    stopf("'partial' cannot be used together with align='center'")
+  if (is.list(x) && length(unique(lengths(x)))!=1L)
+    stopf("'partial' does not support variable length of columns in 'x'")
+  if (!(is.numeric(n) || (is.list(n) && all(vapply(n, is.numeric, FALSE)))))
+    stopf("n must be integer vector or list of integer vectors")
+  len = if (is.list(x)) length(x[[1L]]) else length(x)
+  verbose = getOption("datatable.verbose")
+  if (verbose)
+    cat("partial2adaptive: froll partial=TRUE trimming 'n' and redirecting to adaptive=TRUE\n")
+  trimn = function(n, len, align) {
+    n = min(n, len)
```

Comment:
do we need to block `n<0`?

Reply:
Check for negative n must be in the code already, I don't think it is worth to duplicate that check. But surely make sense to ensure there is unit test for negative n and partial so the actual behavior is known and expected.

Reply:
Associated test SGTM

Reply:
necessary in order to `frollsum(1:2, 3, partial=TRUE)` work, when input is already smaller than window size.